### PR TITLE
Fixing extra data

### PR DESCRIPTION
**Pull request by tomusdrw**
On ice for now - requires additional fix in ethcore/parity to work.

Requires: https://github.com/ethcore/parity/pull/1060

Resolves: #90 #91 #92

### CODE REVIEW

File: client/components/MiningSettings/index.js [hunk 1]
```diff
@@ -17,7 +16,14 @@ export default class MiningSettings extends Component {
...
-    let onExtraDataChange = (newVal) => {
+    let onExtraDataChange = (newVal, isResetToDefault) => {
+      if (isResetToDefault) {
+        // Ignore input value - just set raw bytes
+        actions.modifyExtraData(mining.defaultExtraData);
+        return;
+      }
+
+      // Set string, it will be converted to hex
       actions.modifyExtraData(newVal);
```
Comment (goldylucks) [inline client/components/MiningSettings/index.js]: IMO one of these will b more readabe;
```
// 1
newVal = isResetToDefault ? mining.defaultExtraData : newVal
```
```
// 2
if (isResetToDefault) { 
  newVal = mining.defaultExtraData;
}
```
// 3
const _newVal = isResetToDefault ? mining.defaultExtraData : newVal
actions.modifyExtraData(_newVal)
```

File: client/provider/web3-provider.js [hunk 2]
```diff
@@ -114,13 +115,16 @@ export class Web3Provider extends Web3Base {
...
       if (isBigNumber(val)) {
         return !val.equals(action.payload);
-      } if (isArray(val) || isObject(val)) {
-        return !eq(val, action.payload);
-      } else {
-        return val !== action.payload;
       }
+
+      if (isArray(val) || isObject(val)) {
+        return !isEqual(val, action.payload);
```
Comment (goldylucks) [inline client/provider/web3-provider.js]: isEqual on objects/arrays ?

File: test/mocked-responses.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "rpc": [
     { "name": "eth_coinbase", "response":  "0x0037a6b811ffeb6e072da21179d11b1406371c63" },
-    { "name": "ethcore_extraData", "response":  "0xd783010000867061726974798b312e302e302d6265746132", "formattedResponse": "parity/1.0.0-beta2" },
+    { "name": "ethcore_extraData", "response":  "0xd783010000867061726974798b312e302e302d6265746132", "formattedResponse": "parity/1.0.0/1.0.0-beta2" },
```
Comment (goldylucks) [inline test/mocked-responses.json]: "parity/1.0.0/1.0.0-beta2"  -> copypaste mistake?

File: client/components/MiningSettings/index.js (hunk 1 above)
Comment (tomusdrw) [inline client/components/MiningSettings/index.js]: 1,2 - I'm not really a fan of re-assignment (prefer to keep everything `const`)
3 - that one is fine. Will refactor to that.

File: client/provider/web3-provider.js (hunk 2 above)
Comment (tomusdrw) [inline client/provider/web3-provider.js]: Yes, previous one was actually incorrect - events were triggered for each RPC call that returned array/object.
`eq` - https://lodash.com/docs#eq (Performs a SameValueZero comparison between two values to determine if they are equivalent.)
`isEqual` - https://lodash.com/docs#isEqual (Performs a deep comparison between two values to determine if they are equivalent. )

File: test/mocked-responses.json (hunk 3 above)
Comment (tomusdrw) [inline test/mocked-responses.json]: It's corrected now - previous decoding algorithm was wrong.
We were skipping first element of RLP list, but actually it's encoded parity version (1.0.0); 1.0.0-beta2 is compiler version.

**Pull request merged**